src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -52,6 +52,36 @@ void align_u4_zp_constants(const std::shared_ptr<ov::Model>& model) {
     }
 }
 
+bool allow_to_enable_npuw_dq(const std::shared_ptr<ov::Model>& model) {
+    std::vector<std::string> rt_info_path = {"nncf", "weight_compression", "group_size"};
+    if (!model->has_rt_info(rt_info_path)) {
+        // NB: Model isn't compressed by NNCF - skip
+        return false;
+    }
+    auto group_size = model->get_rt_info<int>(rt_info_path);
+    if (group_size == -1) {
+        // NB: Enable DQ for CW quantized models
+        return true;
+    }
+    return false;
+}
+
+std::optional<ov::Any> pop_option(ov::AnyMap& config, const std::string& option_name) {
+    if (auto it = config.find(option_name); it != config.end()) {
+        config.erase(it);
+        return std::make_optional(it->second);
+    }
+    return std::nullopt;
+}
+
+void enable_npuw_dq_if_allowed(ov::AnyMap& config,
+                               const std::shared_ptr<ov::Model>& model) {
+    if (allow_to_enable_npuw_dq(model)) {
+        config["NPUW_DQ"] = "YES";
+        pop_option(config, "NPUW_ONLINE_AVOID");
```

Comment:
AVOIDs shouldn't be required in general, I'd recommend to remove this as part of the next config update (but in scope for 2024.5)

Reply:
Scheduled